app/controllers/home_controller.rb

```diff
@@ -6,7 +6,7 @@ def top
         year = nowDate.year
         month = nowDate.month
         season, @ranks = @ranks.GetSeasonRecord(year,month)
-        @ranks = @ranks.order(:result)
+        @ranks = @ranks.RankBestOnly.order(:result)
```

Comment:
RankBestOnlyを書くところを変えたらいけると思います！
```suggestion
        @ranks = @ranks.order(:result)
```

Reply:
@topranks.push(@ranks.RankDungeonChoose(dungeon.id).RankBestOnly.limit(3))